JustSaying.Models/Message.cs

```diff
@@ -8,11 +8,20 @@ protected Message()
         {
             TimeStamp = DateTime.UtcNow;
             Id = Guid.NewGuid();
+            CorrelationId = Id;
```

Comment:
Yep, please check with other groups before going down this route - there is likely already such an Id in many http requests, and I don't know if it is a guid.